### ISSUE TITLE
[no-unused-vars] Variables used in an ambient declaration are not detected

### ISSUE THREAD

**Issue by andreubotella** [typescript-eslint/typescript-eslint#2026]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
```

```TS
import { Octokit } from "@octokit/rest";

declare global {
  namespace Express {
    interface Request {
      githubApi?: Octokit;
    }
  }
}
```

**Expected Result**

No ESLint problems are detected.

**Actual Result**

An error for `@typescript-eslint/no-unused-vars` shows up: "'Octokit' is defined but never used". If the `Octokit` type is used elsewhere other than in an ambient declaration, the error goes away.

**Additional Info**

This might be a case covered in the meta-issue #1856, but I don't know enough about the internals of typescript-eslint to know for sure.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.33.0` |
| `@typescript-eslint/parser`        | `2.33.0` |
| `TypeScript`                       | `3.9.2` |
| `ESLint`                           | `7.0.0` |
| `node`                             | `12.16.3` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
#1856 will solve it
